Redesign tables on mobile

jadebuckwalter opened the pull request:
Create a table for assignments without stats that shows your score, the
maximum score, date assigned, date due, and assignment feedback.
For now, I'm trying to get more data to show up in the modal to remove
table columns on mobile, but eventually, the info in the modal will be
transferred to a separate region under the row in the table.

jadebuckwalter committed:
Show score, release/due date on all assignments

Create a table for assignments without stats that shows your score, the
maximum score, date assigned, date due, and assignment feedback.
For now, I'm trying to get more data to show up in the modal to remove
table columns on mobile, but eventually, the info in the modal will be
transferred to a separate region under the row in the table.

jadebuckwalter committed:
Simplify assignments table on mobile

Remove the category, score, and max score columns on mobile so that it's
just the assignment name and percentage.

jadebuckwalter commented:
Still left to do:
- [ ] Add category dropdown to stats modals on mobile
- [ ] Adjust other tables (grades, categories, schedule) for mobile
- [ ] Make the adjustments dynamic
- [ ] Move info in stats modals to an expanded area in the assignments table

jadebuckwalter committed:
Lengthen the modals

I had made the modals smaller in an earlier commit to get rid of the
white space at the bottom; now I realize that if they're too small,
sometimes the table will go off the edge.

jadebuckwalter committed:
Reformat grades table

Decrease the width of the grade column on mobile so that more of the
class name appears.

psvenk reviewed (commented):
This looks great on the whole. I agree that this is in a mergeable state and that the additional changes should be made down the line.

One point of comment: I see that in the case that there are no assignment statistics (but the basic info — score, max score, comments — exists), there is no message saying "there are no stats". Would it be a good idea to alert the user of the fact that there are no assignment statistics, but this is some expanded information about the assignment?

Additionally, the tooltip on the info icon should be changed from "Statistics" to "Info" to indicate its true purpose.

jadebuckwalter committed:
Change stats/info modal height to "auto"

Rather than setting a specific height for the  modals, make them adjust
to the height of the content.

jadebuckwalter committed:
Change "statistics" tooltip to "info"

This is a better label because not all assignments have statistics.
Furthermore, it fits better with the icon.

jadebuckwalter committed:
Set minimum info modal height in CSS file

Instead of setting the height separately in the JS file, set a minimum
in the CSS file and default to "auto."

jadebuckwalter committed:
Update loading message for assignment info

Since the stats modals now contain more information if statistics are
unavailable, the "loading" message has been updated to reflect this.

psvenk reviewed (approved):
Looks good to me on the whole once again; thanks for the additional changes. Another issue that I noticed is that the media matching is done only once, at the page load. Is there any way (perhaps via an event listener) to make it dynamic, so that resizing a browser window or rotating a phone would cause the table to adapt?

I suppose that we can address this in a separate pull request, along with the other areas that you identified, so this should be ready to merge.

Edit: Never mind; I just realized that you already identified this issue when you wrote "make the adjustments dynamic".
